tests/TestCase/Network/ResponseTest.php

```diff
@@ -2778,6 +2778,52 @@ public function testWithBody()
     }
 
     /**
+     * Test with string body.
+     *
+     * @return void
+     */
+    public function testWithStringBody()
+    {
+        $response = new Response();
+        $response->withStringBody('Foo');
+        $body = $response->getBody();
+        $this->assertSame('Foo', (string)$body);
+
+        $response = new Response();
+        $response->withStringBody('');
+        $body = $response->getBody();
```

Comment:
These tests are checking the wrong response object. They should be checking the return value of `withStringBody()` like:

```php
$response = new Response();
$response = $response->withStringBody('');
$body = $response->getBody();
$this->assertSame('', (string)$body);
```

Right now your tests are checking the original response which is not how the immutable style methods should work.